pandas/tests/io/pytables/test_store.py

```diff
@@ -1708,11 +1708,16 @@ def col(t, column):
                 df = tm.makeTimeDataFrame()
                 df["string"] = "foo"
                 df["string2"] = "bar"
+                df["string3"] = "foobar"
                 store.append("f", df, data_columns=["string", "string2"])
                 assert col("f", "index").is_indexed is True
                 assert col("f", "string").is_indexed is True
                 assert col("f", "string2").is_indexed is True
 
+                msg = "'Cols' object has no attribute 'string3'"
```

Comment:
can you make a new test which illustrates this issue (much easier to follow that way)